Test Spanish translation via babeldown (#95)
epiverse-trace/tutorials-middle: locale/es/episodes/severity-static.Rmd

estimación de CFR sólo utilizando los datos observados hasta el 27 Mar 2003
([Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852))](fig/cfr-pone.0006852.g003-fig_c.png)

::::::::::::::::::::::: instructor

Los periodos son relevantes: Periodo 1 -- 15 días en los que el CFR es cero para indicar que se debe a que no se han notificado muertes; Periodo del 15 de marzo -- 26 de abril en el que el CFR parece aumentar; Periodo del 30 de abril -- 30 de mayo en el que la estimación del CFR se estabiliza.

:::::::::::::::::::::::

En términos más generales, estimar la gravedad puede ser útil incluso fuera de un escenario de planificación pandémica y en el contexto de la salud pública rutinaria.
Saber si un brote tiene o tuvo una gravedad diferente a la del registro histórico puede motivar investigaciones causales,
que podrían ser intrínsecas al agente infeccioso (por ejemplo, una nueva cepa más grave) o debidas a factores subyacentes en la población (por ejemplo, inmunidad reducida o factores de morbilidad) ([Lipsitch et al., 2015](https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0003846)).

> En este tutorial vamos a aprender a utilizar la función `{cfr}` para calcular y ajustar una estimación CFR utilizando [distribuciones de retardo](../learners/reference.md#delaydist) de `{epiparameter}` o de otro lugar, basándose en los métodos desarrollados por [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) además, cómo podemos reutilizar `{cfr}` funciones para más mediciones de gravedad.

Reviewer: _(...) [Nishiura et al., 2009](https://journals.plos.org/plosone/articleid=10.1371/journal.pone.0006852)_. **Además, aprenderemos cómo reimplementar las funciondes de `{cfr}` para realizar otras mediciones de gravedad.**

> En este tutorial vamos a aprender a utilizar la función `{cfr}` para calcular y ajustar una estimación CFR utilizando [distribuciones de retardo](../learners/reference.md#delaydist) de `{epiparameter}` o de otro lugar, basándose en los métodos desarrollados por [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) además, cómo podemos reutilizar `{cfr}` funciones para más mediciones de gravedad.

Reviewer 2: vale lo he añadido y con un ejemplo tras consultarlo con @avallecam para que sea mas claro

Utilizaremos la tubería `%>%` para conectar funciones, así que vamos a llamar también a la función `{tidyverse}` paquete:

```{r, message=FALSE, warning=FALSE}
library(cfr)
library(epiparameter)
library(tidyverse)
library(outbreaks)
```

::::::::::::::::::: lista de comprobación

### El doble punto

El doble punto `::` en R te permite llamar a una función específica de un paquete sin cargar todo el paquete en el entorno actual.

Por ejemplo `dplyr::filter(data, condition)` utiliza `filter()` del `{dplyr}` paquete.

> Esto nos ayuda a recordar las funciones del paquete y a evitar conflictos de espacio de nombres.

Reviewer: De acuerdo, pero creo que namespace puede seguir siendo confuso. Propongo ser más explícitos:

**Esto nos ayuda a recordar las funciones requeridas y evitar usar funciones que fueron creadas con el mismo nombre pero provienen de otros paquetes.**

> Esto nos ayuda a recordar las funciones del paquete y a evitar conflictos de espacio de nombres.

Reviewer 2: ```suggestion
Esto nos ayuda a recordar las funciones requeridas y evitar usar funciones que fueron creadas con el mismo nombre pero provienen de otros paquetes.
```

> Esto nos ayuda a recordar las funciones del paquete y a evitar conflictos de espacio de nombres.

Reviewer 2: ok añadido

```

El CFR ajustado al retraso indicó que la gravedad general de la enfermedad *al final del brote* o con el *últimos datos disponibles en ese m* es `r out_estimate` con un intervalo de confianza del 95% entre `r out_low` y `r out_high` ligeramente superior al ingenuo.

:::::::::::::::::: callout

### Utiliza la clase epidist

Cuando utilices una clase `<epidist>` podemos utilizar esta expresión como plantilla:

`function(x) density(<EPIDIST_OBJECT>, x)`

Para funciones de distribución con parámetros no disponibles en `{epiparameter}` te proponemos dos alternativas:

- Crear una `<epidist>` objeto de clase, para conectarlo a otros paquetes R de la tubería de análisis de brotes. Lee la [documentación de referencia de `epiparameter::epidist()`](https://epiverse-trace.github.io/epiparameter/reference/epidist.html) o

- Lee `{cfr}` viñeta para [una introducción al trabajo con distribuciones de retardo](https://epiverse-trace.github.io/cfr/articles/delay_distributions.html).

::::::::::::::::::

:::::::::::::::::::::::::::::::::::::::: challenge

Utiliza el mismo archivo del reto anterior ([sarscov2\_casos\_muertes.csv](data/sarscov2_cases_deaths.csv)).

> Estima el CFR ajustado al retraso utilizando el retraso de distribución apropiado. Entonces:

Reviewer 2: ```suggestion
Estima la TL ajustada al retraso temporal utilizando la distribución apropiada y
```

> - ¡Compara las soluciones CFR ingenuas y las ajustadas al retraso!

Reviewer 2: ```suggestion
- ¡Compara los valores de la TL ajustada y sin ajuste temporal!
```

:::::::::::::::::::: pista

- Encuentra el `<epidist>` ¡apropiado!

::::::::::::::::::::

:::::::::::::::::::: solución

> Utilizamos `{epiparameter}` para acceder a una distribución de retrasos para los datos de incidencia agregados del SARS-CoV-2:

Reviewer 2: ```suggestion
Utilizamos `{epiparameter}` para acceder a una distribución temporal para los datos de incidencia agregados del SARS-CoV-2:
```

```{r, message=FALSE, warning=FALSE}
library(epiparameter)

sarscov2_delay <-
  epidist_db(
    disease = "covid",
    epi_dist = "onset to death",
    single_epidist = TRUE
  )
```

> Leemos la entrada de datos mediante `readr::read_csv()`. Esta función reconoce que la columna `date` es una `<date>` vector de clase.

Reviewer 2: ```suggestion
Leemos el archivo de datos mediante `readr::read_csv()`. Esta función reconoce que la columna `date` es de la clase `<date>`, es decir, una fecha.
```

```{r, eval=TRUE, echo=FALSE, warning=FALSE, message=FALSE}
# read data from the tutorial repository R project
sarscov2_input <-
  readr::read_csv(file.path("data", "sarscov2_cases_deaths.csv"))
```

```{r, eval=FALSE, echo=TRUE}
# read data
# e.g.: if path to file is data/raw-data/ebola_cases.csv then:
sarscov2_input <-
  readr::read_csv(here::here("data", "raw-data", "sarscov2_cases_deaths.csv"))
```

```{r}
# Inspect data
sarscov2_input
```

> Podemos utilizar `dplyr::rename()` para adaptar los datos externos a la entrada de datos de `cfr_static()`.

Reviewer 2: ```suggestion
Podemos utilizar `dplyr::rename()` para cambiar los nombres de las columnas del archivo a los requeridos por`cfr_static()`.
```

```{r}
# Rename before Estimate naive CFR
sarscov2_input %>%
  dplyr::rename(
    cases = cases_jpn,
    deaths = deaths_jpn
  ) %>%
  cfr::cfr_static(delay_density = function(x) density(sarscov2_delay, x))
```

> Interpreta la comparación entre las estimaciones CFR ingenuas y las ajustadas al retraso.

Reviewer 2: ```suggestion
Interpreta las diferencias entre las estimaciones de la tasa de letalidad sin ajustar y con ajuste temporal al intervalo entre la aparición inicial de síntomas en los casos y su muerte.
```

::::::::::::::::::::

::::::::::::::::::::::::::::::::::::::::

:::::::::::::::::: spoiler

### ¿Cuándo utilizar distribuciones discretas?

> En `cfr_static()` y todas las `cfr_*()` familia de funciones, la opción más adecuada para pasar son **discretas** discretas. Esto se debe a que trabajaremos con datos diarios de casos y defunciones.

Reviewer 2: ```suggestion
En `cfr_static()` y toda la familia de funciones en `cfr_*()`, la opción más adecuada son las distribuciones **discretas**. Esto se debe a que trabajaremos con datos diarios de casos y defunciones.
```

> Podemos suponer que la evaluación de la Función de Distribución de Probabilidades (FDP) de una *continua* es equivalente a la función de masa de probabilidad (FMP) de la distribución *discreta* discreta equivalente.

Reviewer 2: ```suggestion
Podemos suponer que la función de densidad de probabilidad (PDF) de una distribución *continua* es equivalente a la función de masa de probabilidad (PMF) de la distribución *discreta* correspondiente.
```

> Sin embargo, esta suposición puede no ser adecuada para distribuciones con picos más grandes. Por ejemplo, las enfermedades con una distribución de inicio-muerte que tiene picos fuertes con una varianza baja. En tales casos, se espera que la disparidad media entre la PDF y la PMF sea más pronunciada en comparación con las distribuciones con picos más amplios. Una forma de abordar esto es discretizar la distribución continua utilizando `epiparameter::discretise()` a una `<epidist>` objeto.

Reviewer 2: ```suggestion
Sin embargo, esta suposición puede no ser adecuada en el caso de distribuciones con picos pronunciados y baja varianza, en las que los valores se concentran alrededor de la tendencia central.  En tales casos, se espera que la disparidad media entre la PDF y la PMF sea más pronunciada en comparación con las distribuciones más dispersas. Una forma de abordar esto es discretizar la distribución continua. Esto es fácil de hacer en R, si discretizamos los objetos `<epidist>`, utilizando la función `epiparameter::discretise()`.
```

::::::::::::::::::

::::::::::::::::::::::::::: spoiler

### ¿Cómo {cfr} funciona?

> Para ajustar el CFR, [Nishiura y otros, 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) utilizan los datos de incidencia de casos y muertes para estimar el número de casos con resultados conocidos:

Reviewer 2: ```suggestion
Para ajustar la tasa de letalidad, [Nishiura y otros, 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) utilizan los datos de incidencia de casos y muertes para estimar el número de casos con resultados conocidos:
```

$$
u_t = \dfrac{\sum_{i = 0}^t
\sum_{j = 0}^\infty c_{i - j} f_{j}}{\sum_{i = 0} c_i},
$$

donde

- $c_{t}$ es la incidencia diaria de casos en el momento $t$,

> - $f_{t}$ es el valor de la Función de Masa Probable (FMP) de la **distribución del retraso** entre el inicio y la muerte, y

Reviewer 2: ```suggestion
- $f_{t}$ es el valor de la función de masa de probabilidad (PMF) de la **distribución temporal** entre el inicio de síntomas y la muerte, y
```

- $u_{t}$ representa el factor de subestimación de los resultados conocidos.

> $u_{t}$ se utiliza para **escalar** el valor del número acumulado de casos en el denominador en el cálculo del CFR. Se calcula internamente con el [`estimate_outcomes()`](https://epiverse-trace.github.io/cfr/reference/estimate_outcomes.html) función

Reviewer 2: ```suggestion
$u_{t}$ se utiliza para **ajustar** el valor del número acumulado de casos en el denominador en el cálculo de la TL. Se calcula internamente con la función [`estimate_outcomes()`](https://epiverse-trace.github.io/cfr/reference/estimate_outcomes.html)
```

> El estimador para CFR puede escribirse como:

Reviewer 2: ```suggestion
El estimador para la tasa de letalidad puede expresarse como:
```

$$p_{t} = \frac{b_{t}}{u_{t}}$$

> donde $p_{t}$ es la proporción realizada de casos confirmados que morirán a causa de la infección (o la CFR insesgada), y $b_{t}$ es la estimación cruda y sesgada de la CFR (también CFR ingenua).

Reviewer 2: ```suggestion
donde $p_{t}$ es la proporción realizada de casos confirmados que morirán a causa de la infección (o la TL real), y $b_{t}$ es la estimación cruda y sesgada de la TL.
```

> A partir de esta última ecuación, observamos que la CFR no sesgada $p_{t}$ es mayor que el CFR sesgado $b_{t}$ porque en $u_{t}$ el numerador es menor que el denominador (observa que $f_{t}$ es la distribución de probabilidad del *distribución del retraso* entre el inicio y la muerte). Por tanto, nos referimos a $b_{t}$ como el estimador sesgado del CFR.

Reviewer 2: ```suggestion
A partir de esta última ecuación, observamos que la TL no sesgada $p_{t}$ es mayor que la TL sesgada $b_{t}$ porque en $u_{t}$ el numerador es menor que el denominador (observa que $f_{t}$ es la distribución de probabilidad del *retraso temporal* entre síntomas y muerte). Por tanto, nos referimos a $b_{t}$ como el estimador sesgado de la tasa de letalidad.
```

> Cuando observamos todo el curso de una epidemia (desde $t \rightarrow \infty$), $u_{t}$ tiende a 1, lo que hace que $b_{t}$ tiende a $p_{t}$ y se convierta en un estimador insesgado ([Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852)).

Reviewer 2: ```suggestion
Cuando observamos todo el curso de una epidemia (desde $t \rightarrow \infty$), $u_{t}$ tiende a 1, lo que hace que $b_{t}$ tiende a $p_{t}$ y se convierta en un estimador no sesgado ([Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852)).
```

:::::::::::::::::::::::::::

> ## Una estimación temprana del CFR

Reviewer 2: ```suggestion
## Estimación temprana de la TL
```

> En el reto anterior, descubrimos que el *ingenuo* y *ajustado al retraso* son diferentes.

Reviewer 2: ```suggestion
En el reto anterior, descubrimos que el valor de la tasa de letalidad ajustada y no ajustada son diferentes.
```

> En **ingenuo** es útil para obtener una estimación global de la gravedad del brote (hasta el momento). Una vez que el brote haya finalizado o haya progresado de forma que se notifiquen más muertes, el CFR estimado es entonces *más cercano a* la "verdadera" CFR no sesgada.

Reviewer 2: ```suggestion
La TL sin ajustar es útil para obtener una estimación global de la gravedad del brote. Una vez que el brote haya finalizado o haya progresado de forma que se notifiquen más muertes, la TL estimada es entonces *más cercana a* la TL "verdadera" o no sesgada.
```

> Por otra parte, el **ajustado al re** puede evaluar la gravedad de una enfermedad infecciosa emergente *antes* que el CFR sesgado o ingenuo, durante una epidemia.

Reviewer 2: ```suggestion
Por otra parte, la TL **ajustada** puede ser utilizada para estimar la gravedad de una enfermedad infecciosa emergente de una forma más temprana durante una epidemia.
```

> Podemos explorar la **temprano** determinación de la *CFR ajustado al retraso* utilizando el `cfr_rolling()` función

Reviewer 2: ```suggestion
Podemos explorar la *TL ajustada al retraso temporal* de forma temprana utilizando la función `cfr_rolling()`
```

:::::::::::::::::::::: callout

> `cfr_rolling()` es una función utilitaria que calcula automáticamente el CFR de cada día del brote con los datos disponibles hasta ese día, ahorrando tiempo al usuario.

Reviewer 2: ```suggestion
`cfr_rolling()` es una función que calcula automáticamente la TL en cada día del brote con los datos disponibles hasta ese día, lo que ahorra tiempo al usuario, ya que no hace falta calcular este parámetro manualmente para cada momento.
```

::::::::::::::::::::::

```{r}
# Calculate the rolling daily naive CFR
# for all the 73 days in the Ebola dataset
rolling_cfr_naive <- cfr::cfr_rolling(data = ebola1976)

utils::tail(rolling_cfr_naive)
```

```{r}
# Calculate the rolling daily delay-adjusted CFR
# for all the 73 days in the Ebola dataset
rolling_cfr_adjusted <- cfr::cfr_rolling(
  data = ebola1976,
  delay_density = function(x) density(onset_to_death_ebola, x)
)

utils::tail(rolling_cfr_adjusted)
```

> Con `utils::tail()` mostramos que las últimas estimaciones de CFR. Las estimaciones ingenuas y las ajustadas al retraso tienen rangos superpuestos de intervalos de confianza del 95%.

Reviewer 2: ```suggestion
Con `utils::tail()` mostramos como los últimos valores estimados de la TL ajustada y sin ajustar tienen rangos superpuestos de intervalos de confianza del 95%.
```

> Ahora, visualicemos ambos resultados en una serie temporal. ¿Cómo se comportarían en tiempo real las estimaciones de CFR ingenuas y las ajustadas al retraso?

Reviewer 2: ```suggestion
Ahora, visualicemos ambos resultados en una serie temporal. ¿Cómo se comportarían en tiempo real las estimaciones de TL ajustadas y sin ajustar?
```

```{r, echo=TRUE, warning=FALSE, message=FALSE}
# get the latest delay-adjusted CFR
rolling_cfr_adjusted_end <-
  rolling_cfr_adjusted %>%
  dplyr::slice_tail()

# bind by rows both output data frames
bind_rows(
  rolling_cfr_naive %>%
    mutate(method = "naive"),
  rolling_cfr_adjusted %>%
    mutate(method = "adjusted")
) %>%
  # visualise both adjusted and unadjusted rolling estimates
  ggplot() +
  geom_ribbon(
    aes(
      date,
      ymin = severity_low,
      ymax = severity_high,
      fill = method
    ),
    alpha = 0.2, show.legend = FALSE
  ) +
  geom_line(
    aes(date, severity_estimate, colour = method)
  ) +
  geom_hline(
    data = rolling_cfr_adjusted_end,
    aes(yintercept = severity_estimate)
  ) +
  geom_hline(
    data = rolling_cfr_adjusted_end,
    aes(yintercept = severity_low),
    lty = 2
  ) +
  geom_hline(
    data = rolling_cfr_adjusted_end,
    aes(yintercept = severity_high),
    lty = 2
  )
```

> La línea horizontal representa la CFR ajustada al retraso estimada al final del brote. La línea punteada significa que la estimación tiene un intervalo de confianza del 95% (IC 95%).

Reviewer 2: ```suggestion
La línea horizontal representa la TL ajustada al retraso temporal, estimada al final del brote. La línea punteada significa que la estimación tiene un intervalo de confianza del 95% (IC 95%).
```

> **Observa** que este cálculo ajustado al retardo es especialmente útil cuando un *curva epidémica de casos confirmados* es el único dato disponible (es decir, cuando no se dispone de datos individuales desde el inicio hasta la muerte, especialmente durante la fase inicial de la epidemia). Cuando hay pocas muertes o ninguna, hay que hacer una suposición para el *distribución del retraso* desde el inicio hasta la muerte, por ejemplo, a partir de la literatura basada en brotes anteriores. [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) representan esto en las figuras con datos del brote de SRAS en Hong Kong, 2003.

Reviewer 2: ```suggestion
**Observa** que este cálculo ajustado al retraso temporal entre síntomas y muerte es especialmente útil cuando los únicos datos disponibles son *curvas epidémica de casos confirmados* (es decir, cuando no se dispone de datos individuales, especialmente durante la fase inicial de la epidemia). Cuando hay pocas muertes o ninguna, hay que hacer una suposición para la *distribución temporal* desde la aparición de síntomas hasta la muerte, por ejemplo, a partir de la literatura basada en brotes anteriores. [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852) representan esto en las figuras con datos del brote de SARS en Hong Kong, 2003.
```

:::::::::::::::::::::::::::::::::: spoiler

> ### Estudio de caso: Brote de SRAS, Hong Kong, 2003

Reviewer: ```suggestion
### Caso de estudio: Brote de SRAS, Hong Kong, 2003
```

> Las figuras A y B muestran el número acumulado de casos y muertes por SRAS, y la figura C muestra las estimaciones observadas (sesgadas) del CFR en función del tiempo, es decir, el número acumulado de muertes sobre casos en el momento $t$. Debido al retraso desde el inicio de los síntomas hasta la muerte, la estimación sesgada de la CFR en el tiempo $t$ subestima el CFR realizado al final de un brote (es decir, 302/1755 = 17,2%).

Reviewer: ```suggestion
Las figuras A y B muestran el número acumulado de casos y muertes por SRAS, y la figura C muestra las estimaciones observadas (sesgadas) del CFR en función del tiempo, es decir, el número acumulado de muertes sobre casos en el momento $t$. Debido al retraso temporal desde el inicio de los síntomas hasta la muerte, la estimación sesgada de la Tasa de Letalidad en el tiempo $t$ subestima el valor en las etapas finales del brote (es decir, 302/1755 = 17,2%).
```

![Riesgo observado (sesgado) de letalidad confirmada del síndrome respiratorio agudo grave (SRAS) en Hong Kong, 2003. ([Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852))](fig/cfr-pone.0006852.g003-fig_abc.png)

> No obstante, incluso utilizando únicamente los datos observados para el periodo comprendido entre el 19 de marzo y el 2 de abril, `cfr_static()` se puede obtener una predicción adecuada (Figura D), por ejemplo, la CFR ajustada al retraso en el 27 de marzo es del 18,1% (IC del 95%: 10,5, 28,1). Se observa una sobreestimación en las fases muy tempranas de la epidemia, pero los límites de confianza del 95% en las fases posteriores incluyen el CFR realizado (es decir, 17,2 %).

Reviewer: ```suggestion
No obstante, incluso utilizando únicamente los datos observados para el periodo comprendido entre el 19 de marzo y el 2 de abril, `cfr_static()` puede obtener una predicción adecuada (Figura D), por ejemplo, la TL ajustada al retraso en el 27 de marzo es del 18,1% (IC del 95%: 10,5, 28,1). Se observa una sobreestimación en las fases muy tempranas de la epidemia, pero los límites de confianza del 95% en las fases posteriores incluyen la TL realizado (es decir, 17,2 %).
```

![Determinación temprana del riesgo de letalidad confirmada ajustado al retraso del síndrome respiratorio agudo grave (SRAS) en Hong Kong, 2003. ([Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852))](fig/cfr-pone.0006852.g003-fig_d.png)

::::::::::::::::::::::::::::::::::

:::::::::::::::::::::::::::::::::::::::::::: debate

> ### Interpretar la estimación del CFR en fase inicial

Reviewer: ```suggestion
### Interpretar la estimación de la TL en fase inicial
```

Basándote en la figura anterior:

> - ¿Cuánta diferencia en días hay entre la fecha en la que el IC del 95% de la estimación *CFR ajustado por retraso* frente a *CFR ingenua* ¿cruzar con el CFR estimado al final del brote?

Reviewer: ```suggestion
- ¿Cuántos días hay entre el inicio del brote y la fecha en la que el intervalo de confianza de la *TL ajustada* se cruza con el intervalo de confianza de la *TL sin ajustar*? ¿Los intervalos se cruzan con la TL estimada al final del brote?
```

> Discute:

Reviewer: ```suggestion
Discusión:
```

> - ¿Cuáles son las implicaciones para la política de salud pública de tener una *MCR ajustado por retraso* ajustada al retraso?

Reviewer: ```suggestion
- ¿Cuáles son las implicaciones para la política de salud pública de tener una *TL ajustada por retraso temporal*?
```

::::::::::::::::::::::::::::::::::::::::::::

:::::::::::::::::::::: sugerencia

Podemos utilizar la inspección visual o el análisis de los marcos de datos de salida.

::::::::::::::::::::::

:::::::::::::::::::::: solución

Hay casi un mes de diferencia.

> Nótese que la estimación tiene una incertidumbre considerable al principio de la serie temporal. Al cabo de dos semanas, el CFR ajustado al retraso se aproxima a la estimación global del CFR al final del brote.

Reviewer: ```suggestion
Nótese que la estimación tiene una incertidumbre considerable al principio de la serie temporal. Al cabo de dos semanas, la TL ajustada se aproxima a la estimación global de la TL al final del brote.
```

¿Es este patrón similar al de otros brotes? Podemos utilizar los conjuntos de datos de los retos de este episodio. ¡Te invitamos a averiguarlo!

::::::::::::::::::::::

:::::::::::::::::::::: debate

> ### Lista de control

Reviewer: ```suggestion
### Lista de verificación
```

> Con `{cfr}` estimamos el CFR como la proporción de muertes entre **confirmadas** confirmados.

Reviewer: ```suggestion
Con `{cfr}` estimamos la TL como la proporción de muertes entre **confirmadas** confirmados.
```

Reviewer: No entiendo la proporción expresada acá

> Utilizando sólo **confirmado** está claro que se pasarán por alto todos los casos que no busquen tratamiento médico o no sean notificados, así como todos los casos asintomáticos. Esto significa que la estimación de la CFR es superior a la proporción de muertes entre los infectados.

Reviewer: ```suggestion
Utilizando sólo el número de casos **confirmados** está claro que se pasarán por alto todos los casos que no busquen tratamiento médico o no sean notificados, así como todos los casos asintomáticos. Esto significa que la estimación de la TL es superior a la proporción de muertes entre los infectados.
```

::::::::::::::::::::::

::::::::::::::::::::::::::: solución

> ### ¿Por qué difieren la ingenua y la ajustada al retraso?

Reviewer: ```suggestion
### ¿Por qué difieren la TL sin ajustar y la ajustada al retraso temporal?
```

> `{cfr}` tiene como objetivo obtener un estimador insesgado "mucho antes" de observar el curso completo del brote. Para ello `{cfr}` utiliza el factor de subestimación $u_{t}$ para estimar el CFR insesgado $p_{t}$ utilizando métodos de máxima verosimilitud, dado el *proceso de muestreo* definido por [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852).

Reviewer: ```suggestion
`{cfr}` tiene como objetivo obtener un estimador insesgado "mucho antes" de observar el curso completo del brote. Para ello `{cfr}` utiliza el factor de subestimación $u_{t}$ para estimar la TL sin sesgo $p_{t}$ , utilizando métodos de máxima verosimilitud, dado el *proceso de muestreo* definido por [Nishiura et al., 2009](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0006852).
```

:::::::::::::::::::::::::::

:::::::::::::::::::::::::: solución

### ¿Qué es el proceso de muestreo?

> ![La población de casos confirmados y el proceso de muestreo para estimar el CFR insesgado durante el transcurso de un brote. ([Nishiura et al., 2009](https://doi.org/10.1371/journal.pone.0006852.g001))](fig/cfr-pone.0006852.g001.png)

Reviewer: ```suggestion
![La población de casos confirmados y el proceso de muestreo para estimar la TL sin sesgo durante el transcurso de un brote. ([Nishiura et al., 2009](https://doi.org/10.1371/journal.pone.0006852.g001))](fig/cfr-pone.0006852.g001.png)
```

> En *datos agregados de incidencia* en el momento $t$ conocemos el número acumulado de casos confirmados y muertes, $C_{t}$ y $D_{t}$ y deseamos estimar el CFR insesgado $\pi$ mediante el factor de subestimación $u_{t}$.

Reviewer: ```suggestion
En *datos agregados de incidencia* en el momento $t$ conocemos el número acumulado de casos confirmados y muertes, $C_{t}$ y $D_{t}$ y deseamos estimar la TL sin sesgo $\pi$ mediante el factor de subestimación $u_{t}$.
```

Si conociéramos el factor de subestimación $u_{t}$ podríamos especificar el tamaño de la población de casos confirmados que ya no corren riesgo ($u_{t}C_{t}$, **sombreado**), aunque no sabemos qué individuos supervivientes pertenecen a este grupo. Una proporción $\pi$ de los del grupo de casos aún en riesgo (tamaño $(1- u_{t})C_{t}$, **sin sombrear**) se espera que muera.

> Porque cada caso que deja de estar en riesgo tiene una probabilidad independiente de morir, $\pi$ el número de muertes, $D_{t}$ es una muestra de una distribución binomial con tamaño de muestra $u_{t}C_{t}$ y probabilidad de morir $p_{t}$ = $\pi$.

Reviewer: ```suggestion
Ya que cada caso que deja de estar en riesgo tiene una probabilidad independiente de morir, $\pi$ el número de muertes, $D_{t}$ es una muestra de una distribución binomial con tamaño de muestra $u_{t}C_{t}$ y probabilidad de morir $p_{t}$ = $\pi$.
```

> Esto se representa mediante la siguiente función de verosimilitud para obtener la estimación de máxima verosimilitud del CFR insesgado $p_{t}$ = $\pi$:

Reviewer: ```suggestion
Esto se representa mediante la siguiente función de verosimilitud para obtener la estimación de máxima verosimilitud de la TL sin sesgo $p_{t}$ = $\pi$:
```

$$
{\sf L}(\pi | C_{t},D_{t},u_{t}) = \log{\dbinom{u_{t}C_{t}}{D_{t}}} + D_{t} \log{\pi} +
(u_{t}C_{t} - D_{t})\log{(1 - \pi)},
$$

Esta estimación la realiza la función interna `?cfr:::estimate_severity()`.

::::::::::::::::::::::::::

:::::::::::::::::::::::::: solución

### Limitaciones

> - El CFR ajustado al retraso no aborda todas las fuentes de error en los datos, como el infradiagnóstico de individuos infectados.

Reviewer: ```suggestion
- La TL ajustada al retraso temporal no aborda todas las fuentes de error en los datos, como el infradiagnóstico de individuos infectados.
```

::::::::::::::::::::::::::

## Desafíos

::::::::::::::::: callout

> ### Los datos agregados difieren de las listas de líneas

Reviewer: ```suggestion
### Los datos agregados difieren de los listados de casos individuales
```

> *Agregados* Los datos de incidencia difieren de **lista** en los que cada observación contiene datos a nivel individual.

Reviewer: ```suggestion
Los datos de incidencia *agregados*  difieren de **lista** en los que cada observación contiene datos a nivel individual.
```

```{r}
outbreaks::ebola_sierraleone_2014 %>% as_tibble()
```

:::::::::::::::::

> :::::::::::::::::::::::::::::::::: challenge

Reviewer: ```suggestion
:::::::::::::::::::::::::::::::::: desafío
```

> ### Utiliza la incidencia2 para reordenar tus datos

Reviewer: ```suggestion
### Utiliza el paquete incidence2 para reordenar tus datos
```

> A partir de la `{outbreaks}` paquete, carga la lista de casos MERS del `mers_korea_2015` objeto.

Reviewer: ```suggestion
Usando el paquete`{outbreaks}`, carga el listado de casos "MERS" incluido en el objeto `mers_korea_2015`.
```

Reordena esta lista de casos para que encaje en el paquete `{cfr}` entrada del paquete.

> Estima el HFR ajustado al retardo utilizando el retardo de distribución correspondiente.

Reviewer: El HFR parece que no se explica anteriormente

::::::::::::::::: sugerencia

**¿Cómo reorganizar mis datos de entrada?**

> Reorganizar los datos de entrada para el análisis de datos puede llevarte la mayor parte del tiempo. Para estar listo para el análisis *datos de incidencia agregados* te animamos a utilizar `{incidence2}` ¡!

Reviewer: ```suggestion
Reorganizar los datos de entrada para el análisis de datos puede llevarte la mayor parte del tiempo. Para estar listo para el análisis con *datos de incidencia agregados* te animamos a utilizar `{incidence2}` ¡!
```

> Primero, en el [Comenzar](https://www.reconverse.org/incidence2/articles/incidence2.html) de la viñeta `{incidence2}` explora cómo utilizar `date_index` al leer una lista de líneas con fechas en varias columnas.

Reviewer: ```suggestion
Primero, en la viñeta [Inicio](https://www.reconverse.org/incidence2/articles/incidence2.html) de `{incidence2}` explora cómo utilizar `date_index` al leer un listado de casos con fechas en varias columnas.
```

> A continuación, consulta `{cfr}` viñeta sobre [Manejo de datos de `{incidence2}`](https://epiverse-trace.github.io/cfr/articles/data_from_incidence2.html) sobre cómo utilizar la función `cfr::prepare_data()` de los objetos incidencia2.

Reviewer: ```suggestion
A continuación, consulta la viñeta [Manejo de datos de `{incidence2}`](https://epiverse-trace.github.io/cfr/articles/data_from_incidence2.html) del paquete `{cfr}`, sobre cómo utilizar la función `cfr::prepare_data()` con los objetos incidence2.
```

<!-- cita la entrada howto one lineslist + incidencia2 + conexión cfr -->

:::::::::::::::::

::::::::::::::::: solución

```{r, message=FALSE, warning=FALSE}

> # Load packages

Reviewer: ```suggestion
# Carga los paquetes
```

library(cfr)
library(epiparameter)
library(incidence2)
library(outbreaks)
library(tidyverse)

> # Access delay distribution

Reviewer: ```suggestion
# Accede a la distribución de retrasos temporales
```

mers_delay <-
  epidist_db(
    disease = "mers",
    epi_dist = "onset to death",
    single_epidist = TRUE
  )

> # Read linelist

Reviewer: ```suggestion
# Lee el listado de casos
```

mers_korea_2015$linelist %>%
  as_tibble() %>%
  select(starts_with("dt_"))

> # Use {incidence2} to count daily incidence

Reviewer: ```suggestion
# Usa {incidence2} para contabilizar la incidencia diaria
```

mers_incidence <- mers_korea_2015$linelist %>%

>   # converto to incidence2 object

Reviewer: ```suggestion
  # convertir a un objeto incidence2
```

  incidence(date_index = c("dt_onset", "dt_death")) %>%

>   # complete dates from first to last

Reviewer: ```suggestion
  # completa las fechas de la primera a la última
```

  incidence2::complete_dates()

> # Inspect incidence2 output

Reviewer: ```suggestion
# Explora los resultados de incidence2
```

mers_incidence

> # Prepare data from {incidence2} to {cfr}

Reviewer: ```suggestion
# Prepara los datos {incidence2} para usarlos en {cfr}
```

mers_incidence %>%
  prepare_data(
    cases_variable = "dt_onset",
    deaths_variable = "dt_death"
  )

> # Estimate delay-adjusted CFR

Reviewer: ```suggestion
# Estimata la TL ajustada a retrasos temporales
```

mers_incidence %>%
  cfr::prepare_data(
    cases_variable = "dt_onset",
    deaths_variable = "dt_death"
  ) %>%
  cfr::cfr_static(delay_density = function(x) density(mers_delay, x))
```

:::::::::::::::::

::::::::::::::::::::::::::::::::::

> :::::::::::::::::::::::::::::::::::::::::::::::::: challenge

Reviewer: ```suggestion
:::::::::::::::::::::::::::::::::::::::::::::::::: desafío
```

### Heterogeneidad de la gravedad

> La CFR puede diferir entre poblaciones (por ejemplo, edad, espacio, tratamiento); cuantificar estas heterogeneidades puede ayudar a dirigir los recursos adecuadamente y a comparar distintos regímenes asistenciales ([Cori et al., 2017](https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371)).

Reviewer: ```suggestion
La TL puede diferir entre poblaciones (por ejemplo, edad, espacio, tratamiento); cuantificar estas heterogeneidades puede ayudar a dirigir los recursos adecuadamente y a comparar distintos regímenes asistenciales ([Cori et al., 2017](https://royalsocietypublishing.org/doi/10.1098/rstb.2016.0371)).
```

> Utiliza el `cfr::covid_data` marco de datos para estimar una CFR ajustada al retraso estratificada por países.

Reviewer: ```suggestion
Utiliza la base de datos `cfr::covid_data` para estimar una TL ajustada al retraso temporal estratificada por países.
```

::::::::::::::::::::::::: sugerencia

> Una forma de hacer un *análisis estratificado* es aplicar un modelo a datos anidados. Este [`{tidyr}` viñeta](https://tidyr.tidyverse.org/articles/nest.html#nested-data-and-models) te muestra cómo aplicar la `group_by()` + `nest()` a los datos anidados, y luego `mutate()` + `map()` para aplicar el modelo.

Reviewer: ```suggestion
Una forma de hacer un *análisis estratificado* es aplicar un modelo a datos anidados. Esta [viñeta `{tidyr}`](https://tidyr.tidyverse.org/articles/nest.html#nested-data-and-models) te muestra cómo aplicar la función `group_by()` + `nest()` a los datos anidados, y luego `mutate()` + `map()` para aplicar el modelo.
```

:::::::::::::::::::::::::

::::::::::::::::::::::::: solución

```{r, message=FALSE, warning=FALSE}
library(cfr)
library(epiparameter)
library(tidyverse)

covid_data %>% glimpse()

delay_onset_death <-
  epidist_db(
    disease = "covid",
    epi_dist = "onset to death",
    single_epidist = TRUE
  )

covid_data %>%
  group_by(country) %>%
  nest() %>%
  mutate(
    temp =
      map(
        .x = data,
        .f = cfr::cfr_static,
        delay_density = function(x) density(delay_onset_death, x)
      )
  ) %>%
  unnest(cols = temp)
```

¡Estupendo! Ahora puedes utilizar un código similar para cualquier otro análisis estratificado, como la edad, las regiones u otros.

Pero, ¿cómo podemos interpretar que existe una variabilidad de gravedad por países a partir del mismo patógeno diagnosticado?

> Factores locales como la capacidad de análisis, la definición de caso y el régimen de muestreo pueden afectar a la notificación de casos y muertes, afectando así a la comprobación de los casos. Echa un vistazo a la `{cfr}` viñeta sobre [Estimar la proporción de casos que se determinan durante un brote](https://epiverse-trace.github.io/cfr/articles/estimate_ascertainment.html) ¡!

Reviewer: ```suggestion
Factores locales como la capacidad de análisis, la definición de caso y el régimen de muestreo pueden afectar a la notificación de casos y muertes, afectando así a la comprobación de los casos. Echa un vistazo a la viñeta [Estimar la proporción de casos que se determinan durante un brote](https://epiverse-trace.github.io/cfr/articles/estimate_ascertainment.html) del paquete `{cfr}` para más información ¡!
```

:::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::::::::::::::::

## Apéndice

> La página `{cfr}` tiene una función llamada `cfr_time_varying()` con una funcionalidad distinta de `cfr_rolling()`.

Reviewer: ```suggestion
El paquete `{cfr}` tiene una función llamada `cfr_time_varying()` con una funcionalidad distinta de `cfr_rolling()`.
```

::::::::::::::::: callout

### ¿Cuándo utilizar cfr\_rolling()?

> **cfr\_desplazam)** muestra el CFR estimado en cada día del brote, dado que los datos futuros sobre casos y muertes no están disponibles en ese momento. El valor final de *cfr\_rodamient)* estimado es idéntico al de *cfr\_estática()* con los mismos datos.

Reviewer: ```suggestion
**cfr\_rolling)** muestra la TL estimada en cada día del brote, dado que los datos futuros sobre casos y muertes no están disponibles en ese momento. El valor final de *cfr\_rolling)* estimado es idéntico al de *cfr\_static()* con los mismos datos.
```

> Recuerda, como se muestra arriba *cfr\_rodamiento()* es útil para obtener estimaciones de CFR en las primeras fases y comprobar si la estimación de CFR de un brote se ha estabilizado. Así, *cfr\_rodamient)* no es sensible a la duración ni al tamaño de la epidemia.

Reviewer: ```suggestion
Recuerda, como se muestra arriba *cfr\_rolling()* es útil para obtener estimaciones de la TL en las primeras fases y comprobar si la estimación de la TL de un brote se ha estabilizado. Así, *cfr\_rolling()* no es sensible a la duración ni al tamaño de la epidemia.
```

:::::::::::::::::

::::::::::::::::: callout

### ¿Cuándo utilizar cfr\_time\_varying()?

> Por otra parte, **cfr\_tiempo\_variante()** calcula la CFR a lo largo de una ventana móvil y ayuda a comprender los cambios en la CFR debidos a cambios en la epidemia, por ejemplo, debidos a una nueva variante o a una mayor inmunidad por vacunación.

Reviewer: ```suggestion
Por otra parte, **cfr\_time\_varying()** calcula la TL a lo largo de una ventana móvil y ayuda a comprender los cambios en la TL debidos a cambios en la epidemia, por ejemplo, debidos a una nueva variante o a una mayor inmunidad por vacunación.
```

> Sin embargo, *cfr\_tiempo\_variante()* es sensible a la incertidumbre del muestreo. Por tanto, es sensible al tamaño del brote. Cuanto mayor sea el número de casos con resultados esperados en un día determinado, más estimaciones razonables de la CFR variable en el tiempo obtendremos.

Reviewer: ```suggestion
Sin embargo, *cfr\_time\_variying()* es sensible a la incertidumbre del muestreo. Por tanto, es sensible al tamaño del brote. Cuanto mayor sea el número de casos con resultados esperados en un día determinado, más estimaciones razonables de la TL variable en el tiempo obtendremos.
```

> Por ejemplo, con 100 casos, la estimación del riesgo de mortalidad tendrá, a grandes rasgos, un intervalo de confianza del 95% ±10% de la estimación media (IC binomial). Por tanto, si tenemos >100 casos con resultados esperados *en un día det* podemos obtener estimaciones razonables del CFR variable en el tiempo. Pero si sólo tenemos >100 casos *a lo largo de toda la epidemia* probablemente tengamos que basarnos en **cfr\_rodamiento()** que utiliza los datos acumulados.

Reviewer: ```suggestion
Por ejemplo, con 100 casos, la estimación del riesgo de mortalidad tendrá, a grandes rasgos, un intervalo de confianza del 95% ±10% de la estimación media (IC binomial). Por tanto, si tenemos >100 casos con resultados esperados *en un día determinado* podemos obtener estimaciones razonables de la TL variable en el tiempo. Pero si sólo tenemos >100 casos *a lo largo de toda la epidemia* probablemente tengamos que basarnos en **cfr\_rolling()** que utiliza los datos acumulados.
```

> Te invitamos a leer esto [viñeta sobre la `cfr_time_varying()` función](https://epiverse-trace.github.io/cfr/articles/estimate_time_varying_severity.html).

Reviewer: ```suggestion
Te invitamos a leer esta [viñeta sobre `cfr_time_varying()`](https://epiverse-trace.github.io/cfr/articles/estimate_time_varying_severity.html).
```

:::::::::::::::::

:::::::::::::::::::::::::::::::: debate

### Más medidas de gravedad

Supongamos que necesitamos evaluar la gravedad clínica de la epidemia en un contexto distinto al de los datos de vigilancia, como la gravedad entre los casos que llegan a los hospitales o los casos que recogiste en una encuesta serológica representativa.

> Utilizando `{cfr}` podemos cambiar las entradas para el numerador (`cases`) y el denominador (`deaths`) para estimar más medidas de gravedad, como el Riesgo de Fatalidad por Infección (RFI) o el Riesgo de Fatalidad por Hospitalización (RFH). Podemos seguir esta analogía:

Reviewer: ```suggestion
Utilizando `{cfr}` podemos cambiar las entradas para el numerador **casos** (`cases`) y el denominador **muertes** (`deaths`) para estimar más medidas de gravedad, como el Riesgo de Fatalidad por Infección (RFI) o el Riesgo de Fatalidad por Hospitalización (RFH). Podemos seguir esta analogía:
```

::::::::::::::::::::::::::::::::

:::::::::::::::::::::::::::: solución

### Riesgo de mortalidad por infección y hospitalización

> Si por una *Caso* riesgo de mortalidad (CFR), exigimos:

Reviewer: ```suggestion
Si para la una *Tasa de letalidad* (TL o CFR en inglés), exigimos:
```

> - *caso* datos de incidencia de casos y muertes, con un

Reviewer: ```suggestion
- *Incidencia* de casos y muertes, con una:
```

> - distribución de retraso entre casos y muertes (o una aproximación cercana, como entre el inicio de los síntomas y la muerte).

Reviewer: ```suggestion
- Distribución de retraso temporal entre casos y muertes (o una aproximación cercana, como el tiempo entre inicio de los síntomas y la muerte).
```

> Entonces, la *Infección* (IFR) requiere:

Reviewer: ```suggestion
Entonces, el *Riesgo de Fatalidad por Infección* (RFI) requiere:
```

> - *infección* y los datos de incidencia de muertes, con un

Reviewer: ```suggestion
- Incidencia de *infecciones* y muertes, con una
```

> - distribución de retraso de la exposición a la muerte (o una aproximación cercana).

Reviewer: ```suggestion
- Distribución de retraso temporal desde la infección hasta la muerte (o una aproximación cercana).
```

> Del mismo modo, la *Hospitalización* Fatality Risk (HFR) requiere:

Reviewer: ```suggestion
Del mismo modo, el *Riesgo de Fatalidad por Hospitalización* (RFH) requiere:
```

> - *hospitalización* datos de incidencia de hospitalización y muerte, y un

Reviewer: ```suggestion
- Incidencia de *hospitalizaciones* y muertes, y una:
```

> - distribución del retraso de la hospitalización a la muerte.

Reviewer: ```suggestion
- Distribución del retraso temporal entre la hospitalización y la muerte.
```

- sCFR riesgo sintomático de letalidad,
- sCHR riesgo sintomático de hospitalización,
- mCFR riesgo de caso-fatalidad médicamente atendido,
- mCHR riesgo de hospitalización de casos atendidos médicamente,
- HFR riesgo de hospitalización-fatalidad.

![Diagrama esquemático de los análisis de referencia. Las flechas rojas, azules y verdes indican el flujo de datos desde los casos confirmados por laboratorio de la vigilancia pasiva, los casos diagnosticados clínicamente y los casos confirmados por laboratorio de los cribados activos.](fig/cfr-s41467-020-19238-2-fig_b.png)

::::::::::::::::::::::::::::

:::::::::::::::::::: debate

Basándote en tu experiencia:

- Comparte algún brote anterior en el que hayas participado en su respuesta.

Responde a estas preguntas:

- ¿Cómo evaluaste la gravedad clínica del brote?
- ¿Cuáles fueron las principales fuentes de sesgo?
- ¿Qué hiciste para tener en cuenta el sesgo identificado?
- ¿Qué análisis complementario harías para resolver el sesgo?

::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: keypoints

- Utiliza `{cfr}` para estimar la gravedad

> - Utiliza `cfr_static()` para estimar el CFR global con los últimos datos disponibles.

Reviewer: ```suggestion
- Utiliza `cfr_static()` para estimar la TL global con los últimos datos disponibles.
```

> - Utiliza `cfr_rolling()` para mostrar cuál sería el CFR estimado en cada día del brote.

Reviewer: ```suggestion
- Utiliza `cfr_rolling()` para mostrar cuál sería la TL estimada en cada día del brote.
```
